### ISSUE TITLE
Publish prefix / suffix variant for text input

### ISSUE THREAD

**Issue by hannalaakso** [alphagov/govuk-frontend#1915]
## What
Publish the prefix / suffix variant for text input contributed by @simonwhatley 🎉 

## Further detail

[Preview](https://deploy-preview-1269--govuk-design-system-preview.netlify.app/components/text-input/#prefixes-and-suffixes) 
Guidance PR: https://github.com/alphagov/govuk-design-system/pull/1269  
Component PR: https://github.com/alphagov/govuk-frontend/pull/1816
Feedback from June 2020 Working Group review: https://github.com/alphagov/govuk-design-system/pull/1269#issuecomment-654282576

## Done when

- [x] Ensure that the variant works on smaller viewports - see [comment](https://github.com/alphagov/govuk-design-system/issues/1330#issuecomment-673511994) for details
- [x] Swap `span` elements for `div` elements (so that they can contain block level elements if needed)
- [x] Consider adding some more stress cases to Frontend review app
- [x] Add automated tests for the variant

**Comment by CharlotteDowns** [comment]
Design Suggestions
--
### Flexible across viewports
@hannalaakso and I discussed if it was possible to have a flexible way of expanding the prefix and suffix areas to fit the content that is placed into them. 

![800px wide view with prefix, input and suffix inline](https://user-images.githubusercontent.com/56260216/90776711-5ab51000-e2f2-11ea-8f2c-6f3fc42c86fe.png)
![320px wide view with prefix and input inline, suffix on second row](https://user-images.githubusercontent.com/56260216/90776718-5c7ed380-e2f2-11ea-8d3b-e8a3b1b4e9bd.png)
![180px wide view with prefix, input and suffix on separate rows](https://user-images.githubusercontent.com/56260216/90776722-5d176a00-e2f2-11ea-8da9-ad5db0af3ad7.png)

This may be hard to achieve and maintain across operating systems and browsers.

### Prefix integrated into input
We also discussed whether it was possible to include the prefix within the input – seen in a Lloyds bank example.

![Lloyds Bank example of a green pound sign in the input field](https://user-images.githubusercontent.com/56260216/90785331-204f7100-e2fa-11ea-828c-be77532b99b6.jpg)
![Mockup of a GOV.UK input with a prefix in the input](https://user-images.githubusercontent.com/56260216/90785365-29404280-e2fa-11ea-902b-e15e81129cb1.png)

This again would need user research to determine whether it improves the usability and if it would adhere to the latest WCAG guidelines.

### Results of conversation
In conclusion the solution Hannah proposed earlier on this card where she has mocked up a 100% width solution for prefixes, inputs and suffixes seems like the most suitable design.
